src/commits/Commit.ts

```diff
@@ -0,0 +1,23 @@
+import ICommitProtectedHeaders from './ICommitProtectedHeaders';
+import ICommitUnprotectedHeaders from './ICommitUnprotectedHeaders';
+
+/**
+ * Represents a single commit
+ */
+export default abstract class Commit {

```

Comment:
Should we make this an interface rather than a class? I have seen typescript problems when extending classes across package boundaries when the dependency versions don't match exactly